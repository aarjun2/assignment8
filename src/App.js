import logo from './logo.svg';
import './App.css';
import { useState } from 'react';
function App() {
  const [todo, setTodo] = useState([]);
  const [val, setVal] = useState('');

  const handleClick = () => {
    if (val.trim() !== '') {
      setTodo([...todo, val]);
      setVal('');
    }
  }

  const markAsCompleted = (index) => {
    const updatedTodos = [...todo];
    updatedTodos[index] = <strike>{updatedTodos[index]}</strike>;
    setTodo(updatedTodos);
  }

  return (
    <>
      <input type="text" value={val} placeholder="Enter a task"onChange={e => setVal(e.target.value)}/>
      <button onClick={handleClick}> Add todo </button>
      <ul>
        {todo.map((task, index) => (
          <li key={index} onClick={() => markAsCompleted(index)}>
            {task}
          </li>
        ))}
      </ul>
    </>
  );
}

export default App;
