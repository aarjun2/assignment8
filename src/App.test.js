import { render, screen, fireEvent } from '@testing-library/react';
import  renderer from 'react-test-renderer'
import App from './App';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

describe("App",() => {
  it("matches snapshot",() => {
    const tree = renderer 
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})// snap shot test

//tested using react library
test('adds a task', () => {
  render(<App />);
  const inputElement = screen.getByPlaceholderText('Enter a task');
  const addButton = screen.getByText('Add todo');

  fireEvent.change(inputElement, { target: { value: 'New Task' } });
  fireEvent.click(addButton);

  expect(screen.getByText('New Task')).toBeInTheDocument();
});

test('markAsCompleted marks a task as completed', () => {
  const { getByText } = render(<App />);
  const input = screen.getByPlaceholderText('Enter a task');
  fireEvent.change(input, { target: { value: 'Task 1' } });
  fireEvent.click(getByText('Add todo'));

  const task = getByText('Task 1');
  fireEvent.click(task);
  const completedTask = getByText('Task 1');
  expect(completedTask.tagName).toBe('STRIKE');
});