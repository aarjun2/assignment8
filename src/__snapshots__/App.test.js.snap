// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
Array [
  <input
    onChange={[Function]}
    placeholder="Enter a task"
    type="text"
    value=""
  />,
  <button
    onClick={[Function]}
  >
     Add todo 
  </button>,
  <ul />,
]
`;
